README.md

```diff
@@ -1,2 +1,60 @@
+
+## How to test Token Exchange Grant Type:
+
+* Download and build the project by executing `maven clean install`
+* Copy the `grant.token.exchange-1.0.0.jar` into <Carbon-Home>/repository/components/dropins directory
+* Add the following configurations to deployment.toml to enable this grant type:
+```toml
+[[oauth.custom_grant_type]]
+name="urn:ietf:params:oauth:grant-type:token-exchange"
+grant_handler="org.wso2.carbon.identity.oauth2.grant.token.exchange.TokenExchangeGrantHandler"
+grant_validator="org.wso2.carbon.identity.oauth2.grant.token.exchange.TokenExchangeGrantValidator"
+[oauth.custom_grant_type.properties]
+IdTokenAllowed=true
+```
+* Start the WSO2 server
+* Register an Identity Provider in WSO2 Identity Server with the configurations of the external Identity Provider.
+* Obtain an access token from external Identity Provider
+* Execute the following curl command:
+
+```curl
+curl --location --request POST 'https://localhost:9443/oauth2/token?scope=openid' \
+--header 'Content-Type: application/x-www-form-urlencoded' \
+--header 'Authorization: Basic ${base64(clientId:clientSecret)}' \
+--data-urlencode 'grant_type=urn:ietf:params:oauth:grant-type:token-exchange' \
+--data-urlencode 'subject_token=${externalIdPToken}' \
+--data-urlencode 'subject_token_type=urn:ietf:params:oauth:token-type:jwt' \
+--data-urlencode 'requested_token_type=urn:ietf:params:oauth:token-type:jwt'
+``` 
+
+* You will get JWT type access token in the response. Sample response:
+
+```json
+{
+    "access_token": "eyJ4NXQiOiJPVGt3TnpSa01tTmxZekZoWWpFeU56VTNOemN3TjJZNU9EQmpNV1kzTTJJMk1EZGhabUU1TmpCbE1qRmtaR0kxTkdFNU9XVTRPREU0TlRCaE1EWXhZUSIsImtpZCI6Ik9Ua3dOelJrTW1ObFl6RmhZakV5TnpVM056Y3dOMlk1T0RCak1XWTNNMkkyTURkaFptRTVOakJsTWpGa1pHSTFOR0U1T1dVNE9ERTROVEJoTURZeFlRX1JTMjU2IiwiYWxnIjoiUlMyNTYifQ.eyJzdWIiOiJ0ZXN0QGdtYWlsLmNvbSIsImF1dCI6IkFQUExJQ0FUSU9OX1VTRVIiLCJhdWQiOiJYdllmNDI5M2FLX19PNkdXUXV4MDMxZ0VxRUVhIiwibmJmIjoxNjIzMTUyNDQ0LCJhenAiOiJYdllmNDI5M2FLX19PNkdXUXV4MDMxZ0VxRUVhIiwic2NvcGUiOiJkZWZhdWx0IiwiaXNzIjoiaHR0cHM6XC9cL2xvY2FsaG9zdDo5NDQzXC9vYXV0aDJcL3Rva2VuIiwiZXhwIjoxNjIzMTU2MDQ0LCJpYXQiOjE2MjMxNTI0NDQsImp0aSI6IjZmZjJjY2FmLWEzNGMtNDYzZi04MTUxLTUxMDNlNzNkNTljMSJ9.BOxQeP3ZZgckIHazM79AFRdy1-S2ntaCzEDQLQwSQjswuaXHesNARVKrwuyw8v7IJXF_7zFglpF2d9PfoTNwjgpJStW_d-n_1NOUr5eyMU0Y5zHDwOFFKrV51WpV99L1KWZbLwiN_kvUbpVDvTijyNDK29cHxQHak6TqUeqDJfVW92bOAqrp88Rn3h19YRlEttjpPrepKDFJ1lse7gO1NO--87pJpwjWEaniQrNNuB1GDbVXYOLWp5ql-X5w9PrJtnrtbska6sAuURScNL0MpLdB4QiO1cMLVyIedwPotv04qeX80ATv9KOlEkoG2ycNHCmW0iHHspn5HtPyTZxxYg",
+    "refresh_token": "7f3bf5df-feb2-356b-a3fb-0a3fc499bf1e",
+    "issued_token_type": "urn:ietf:params:oauth:token-type:jwt",
+    "scope": "default",
+    "token_type": "Bearer",
+    "expires_in": 3600
+}
+```
+
+---
+**NOTE**
+It supports following configurations in `identity.xml` to enable/disable IAT validation for JWT:
+
+```xml
+<TokenExchangeGrant>
```

Comment:
we need to add this to .j2 template and  the conf should be read from deployment.toml

Reply:
The initial implementation of token exchange is planned to be compatible with IS 5.11. Currently, the j2 changes will not be sent to master (as it is for 5.12). We can send those changes when upgrading the dependencies to be compatible with 5.12.

Reply:
But even in IS 5.11.0 we can't add configurations directly to the identity.xml as during restart the j2 file will override the added configurations which are not templated

Reply:
Yeah, these configs are added to j2 file from APIM side

Reply:
As discussed configs are changed as follows:
```toml
[[oauth.custom_grant_type]]
name="urn:ietf:params:oauth:grant-type:token-exchange"
grant_handler="org.wso2.carbon.identity.oauth2.grant.token.exchange.TokenExchangeGrantHandler"
grant_validator="org.wso2.carbon.identity.oauth2.grant.token.exchange.TokenExchangeGrantValidator"
[oauth.custom_grant_type.properties]
IdTokenAllowed=true
EnableIATValidation=true
IATValidityPeriod=30
```